hail/python/test/hail/matrixtable/test_matrix_table.py

```diff
@@ -1335,6 +1335,29 @@ def test_entry_filter_stats(self):
                                                  fraction_filtered=hl.float32(0.0))})
         assert mt.aggregate_cols(hl.agg.all(mt.entry_stats_col == col_expected[mt.col_idx % 4 == 0]))
 
+    def test_annotate_col_agg_lowering(self):
+        mt = hl.utils.range_matrix_table(10, 10, 2)
+        mt = mt.annotate_cols(c1=[mt.col_idx, mt.col_idx * 2])
+        mt = mt.annotate_entries(e1=mt.col_idx + mt.row_idx, e2=[mt.col_idx * mt.row_idx, mt.col_idx * mt.row_idx ** 2])
+        common_ref = mt.c1[1]
+        mt = mt.annotate_cols(exploded=hl.agg.explode(lambda e: common_ref + hl.agg.sum(e), mt.e2),
+                              array=hl.agg.array_agg(lambda e: common_ref + hl.agg.sum(e), mt.e2),
+                              filt=hl.agg.filter(mt.e1 < 5, hl.agg.sum(mt.e1) + common_ref),
+                              grouped=hl.agg.group_by(mt.e1 % 5, hl.agg.sum(mt.e1) + common_ref))
+        mt.cols()._force_count()
+
+    @pytest.mark.skip("does not pass due to CSE bug")
```

Comment:
What's this? Did you find a bug in the CSE?

Reply:
Yeah, looks like if there's a shared ref in an agg/scan/eval context, it should be getting duplicated for each context but it's not

Reply:
I have a fix ready to go. I'll PR it after this merges.